### PR TITLE
feat(math): implement m:sSup superscript converter (SD-2372)

### LINKED ISSUE #2595: Math: implement m:sSup superscript converter (community)
## What

Implement the `m:sSup` (superscript) OMML → MathML converter.

**OMML**: `m:sSup` → `m:e` (base) + `m:sup` (superscript)  <br>**MathML**: \`\` with two children (base, superscript)  <br>**Spec**: §22.1.2.112  <br>**Difficulty**: Easy

## Example

Word: x² → OMML: `x2`  <br>MathML: `x2`

## How to implement

1. Create `packages/layout-engine/painters/dom/src/features/math/converters/superscript.ts`
2. Export `convertSuperscript: MathObjectConverter`
3. Register in `converters/index.ts` and `omml-to-mathml.ts` registry (`'m:sSup': convertSuperscript`)

See `converters/fraction.ts` for the pattern — \~20 lines of code.

## Test Document

Download the test file and upload it to the SuperDoc dev app to verify your implementation renders correctly:

[sd-2372-superscript.docx](https://uploads.linear.app/92b189d9-e15d-4f28-8a87-d9ca147ba33f/7e26e3b3-be41-4e0c-b907-a49ce0f980c4/b2d65370-56e7-41c1-9acd-75c5ea4e3c7d?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzkyYjE4OWQ5LWUxNWQtNGYyOC04YTg3LWQ5Y2ExNDdiYTMzZi83ZTI2ZTNiMy1iZTQxLTRlMGMtYjkwNy1hNDljZTBmOTgwYzQvYjJkNjUzNzAtNTZlNy00MWMxLTlhY2QtNzVjNWVhNGUzYzdkIiwiaWF0IjoxNzc0NjQ4MjQyLCJleHAiOjE4MDYyMTg4MDJ9.TJc0oQ4nowfOcFereJsyRo4GtYiQEz7GB_FmwjSEo6Q)

---

**Want to work on this?** Check the [Contributing Guide](<https://github.com/superdoc-dev/superdoc/blob/main/CONTRIBUTING.md>) to get started. Comment on this issue to let us know you're picking it up.

*Ref:* [SD-2372](https://linear.app/superdocworkspace/issue/SD-2372/math-implement-mssup-superscript-converter-community)

### PR DESCRIPTION
**Pull request by gpardhivvarma**
## Summary

- Add OMML `m:sSup` → MathML `<msup>` converter following the `fraction.ts` pattern
- Move `m:f` from the "not yet implemented" registry section to "implemented" where it belongs
- 3 tests: basic conversion (x²), properties element ignored, missing `m:sup` graceful degradation

## Test plan

- [x] `pnpm --filter @superdoc/painter-dom test` — `omml-to-mathml.test.ts` passes (17 tests)
- [x] Upload [sd-2372-superscript.docx](https://uploads.linear.app/92b189d9-e15d-4f28-8a87-d9ca147ba33f/7e26e3b3-be41-4e0c-b907-a49ce0f980c4/b2d65370-56e7-41c1-9acd-75c5ea4e3c7d) to dev app and verify superscript renders correctly

Closes #2595

### CODE REVIEW

File: packages/layout-engine/painters/dom/src/features/math/converters/superscript.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ *
+ * @spec ECMA-376 §22.1.2.105
+ */
+export const convertSuperscript: MathObjectConverter = (node, doc, convertChildren) => {
+  const elements = node.elements ?? [];
+  const base = elements.find((e) => e.name === 'm:e');
+  const sup = elements.find((e) => e.name === 'm:sup');
+
+  const msup = doc.createElementNS(MATHML_NS, 'msup');
+  msup.appendChild(convertChildren(base?.elements ?? []));
```
Comment (caio-pizzol) [inline packages/layout-engine/painters/dom/src/features/math/converters/superscript.ts]: when the base or exponent has multiple parts (like `x+1`), they end up as separate children instead of being grouped. same issue as `fraction.ts` — we'll fix all converters together in a follow-up.

not blocking.

**Review by caio-pizzol — COMMENTED** [review]
@gpardhivvarma nice work — clean implementation, spec reference is correct.

simple cases like `x²` work great. compound expressions like `(x+1)²` don't render right — the parts end up flat instead of grouped. same thing happens in the subscript PR and `fraction.ts`, so we'll fix them all together in a follow-up.

nested superscripts, fractions as base, and property handling all work. tests look good.

i made a .docx with 10 variations:

[sd-2372-superscript-edge-cases.docx](https://github.com/user-attachments/files/26361469/sd-2372-superscript-edge-cases.docx)

**Word:**
image
image

**SuperDoc:**
image
image

cases 2, 3, and 7 break (the ones with multiple parts in the base or exponent). the rest render correctly.

File: packages/layout-engine/painters/dom/src/features/math/converters/superscript.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  msup.appendChild(convertChildren(sup?.elements ?? []));
```
[the remark this answers is not in the document]
Comment (caio-pizzol) [reply packages/layout-engine/painters/dom/src/features/math/converters/superscript.ts]: confirmed — we'll fix this across all converters in a follow-up.

[the remark this answers is not in the document]
Comment (gpardhivvarma) [reply packages/layout-engine/painters/dom/src/features/math/converters/superscript.ts]: Fixed — both operands are now wrapped in `<mrow>` before appending, matching the pattern in `bar.ts`. Multi-part expressions like `(x+1)²` now produce exactly 2 children in `<msup>`. Added a test for this case too.

File: packages/layout-engine/painters/dom/src/features/math/converters/superscript.ts (hunk 1 above)
Comment (gpardhivvarma) [reply packages/layout-engine/painters/dom/src/features/math/converters/superscript.ts]: Thanks for the thorough testing! This is now fixed in the latest push — also wrapped `<mrow>` around fraction operands while at it since it had the same issue.

**Review by caio-pizzol — APPROVED** [review]
@gpardhivvarma mrow wrapping is in, fraction fix is a nice bonus. all feedback from last round is addressed.

tested with a 15-case edge case doc (simple, multi-run, nested, cross-converter with fractions and bars, triple nesting, ctrlPr, empty props, sibling superscripts) — all pass.

approving.
